engine/table/src/main/java/io/deephaven/engine/table/impl/select/MatchFilter.java

```diff
@@ -261,10 +265,12 @@ public synchronized void init(
 
     @Override
     public SafeCloseable beginOperation(@NotNull final Table sourceTable) {
-        if (initialDependenciesGathered || dataIndex != null) {
-            throw new IllegalStateException("Inputs already initialized, use copy() instead of re-using a WhereFilter");
+        if (dataIndexGathered) {
```

Comment:
I don't think you should make this change. It's attempting to make permissible the impermissible. Moreover, you're going to `dropReference` the `dataIndex` more than you `retainReference` it with the current implementation. If there are buggy tests that need to `copy()`, we should fix them.